include/seqan3/io/sequence_file/format_genbank.hpp

```diff
@@ -0,0 +1,266 @@
+              typename id_type,
+              typename qual_type>
+    void read(stream_type                                                            	& stream,
+              sequence_file_input_options<seq_legal_alph_type, seq_qual_combined> const & options,
+              seq_type                                                                  & sequence,
+              id_type                                                                   & id,
+              qual_type                                                                 & SEQAN3_DOXYGEN_ONLY(qualities))
+    {
+        auto stream_view = std::ranges::subrange<decltype(std::istreambuf_iterator<char>{stream}),
+                                                 decltype(std::istreambuf_iterator<char>{})>
+                           {std::istreambuf_iterator<char>{stream},
+                            std::istreambuf_iterator<char>{}};
+        auto stream_it = std::ranges::begin(stream_view);
+
+        std::string idbuffer;
+        std::ranges::copy(stream_view | view::take_until_or_throw(is_cntrl || is_blank),
+                          std::back_inserter(idbuffer));
+
+        if (idbuffer != "LOCUS")
+            throw parse_error{"An entry has to start with the code word LOCUS."};
+
+        //ID
+        if constexpr (!detail::decays_to_ignore_v<id_type>)
+        {
+            if (options.embl_genbank_complete_header)
+            {
+                while (idbuffer != "ORIGIN")
+                {
+                        std::ranges::copy(idbuffer  | view::char_to<value_type_t<id_type>>,
+                                     	  std::back_inserter(id));
+                        std::ranges::copy(stream_view | view::take_until_or_throw(is_char<'\n'>)
+                                                	  | view::char_to<value_type_t<id_type>>,
+                                     	  std::back_inserter(id));
+                        std::ranges::copy(std::string_view{"\n"} | view::char_to<value_type_t<id_type>>,
+                                     	  std::back_inserter(id));
+
+                    ++stream_it;
+                    idbuffer.clear();
+                    std::ranges::copy(stream_view | view::take_until_or_throw(is_cntrl || is_blank),
+                                 	  std::back_inserter(idbuffer));
+                }
+            }
+            else
+            {
+				// ID
+                detail::consume(stream_view | view::take_until(!is_blank));
+                // read id
+                std::ranges::copy(stream_view | view::take_until_or_throw(is_cntrl || is_blank)
+                                              | view::char_to<value_type_t<id_type>>,
+                                  std::back_inserter(id));
+                detail::consume(stream_view | view::take_line_or_throw);
+            }
+        }
+
+        // Jump to sequence
+        while (idbuffer != "ORIGIN") //True, if not complete header is read into id
```

Comment:
```suggestion
        while (std::ranges::equal(stream_view | view::take_line_or_throw, std::string{"ORIGIN"})) 
        {}
```
Does everything you want already because it consumes the strem_view again and again until ORIGIN is reached and consumed.